src/wirecraft/client/__init__.py

```diff
@@ -7,154 +7,253 @@
 from .assets import SWITCH_DEVICE
 from .server_interface import ServerInterface
 
-FPS = pygame.time.Clock()
-FPS.tick(30)
 BLACK = (0, 0, 0)
+WHITE = (255, 255, 255)
 RES_LIST = [(3024, 1964), (1920, 1080), (1900, 1200)]
 FLAGS = pygame.FULLSCREEN | pygame.NOFRAME | pygame.SCALED
+FPS = 30
 
 
 @dataclass
 class Camera:
-    x: int
-    y: int
+    x: float
+    y: float
     zoom: float
+    min_zoom: float = 0.3
+    max_zoom: float = 3.0
+
+    def screen_to_world(self, screen_pos: tuple[float, float], screen_size: tuple[int, int]) -> tuple[float, float]:
+        """Convert screen coordinates to world coordinates
+        screen coordinates are relative to the top left corner of the screen, world coordinates are relative to the center of the screen
+
+        Args:
+            screen_pos (tuple[float, float]): the position on the screen
+            screen_size (tuple[int, int]): the size of the screen
+
+        Returns:
+            tuple[float, float]: the position in the world
+        """
+        screen_center = (screen_size[0] / 2, screen_size[1] / 2)
+        rel_x = (screen_pos[0] - screen_center[0]) / self.zoom
+        rel_y = (screen_pos[1] - screen_center[1]) / self.zoom
+        return (rel_x + self.x, rel_y + self.y)
+
+    def world_to_screen(self, world_pos: tuple[float, float], screen_size: tuple[int, int]) -> tuple[float, float]:
+        """Convert world coordinates to screen coordinates
+        screen coordinates are relative to the top left corner of the screen, world coordinates are relative to the center of the screen
+        Args:
+            world_pos (tuple[float, float]): the position in the world
+            screen_size (tuple[int, int]): the size of the screen
+        Returns:
+            tuple[float, float]: the position on the screen
+        """
```

Comment:
```suggestion
        """Convert world coordinates to screen coordinates
        screen coordinates are relative to the top left corner of the screen, world coordinates are relative to the center of the screen

        Args:
            world_pos: the position in the world
            screen_size: the size of the screen

        Returns:
            the position on the screen
        """
```

Add line breaks, and don't repeat argument types as they are defined in the function signature.